site/app/controllers/forum/ForumController.php

```diff
@@ -151,6 +154,34 @@ public function publishPost(){
         }
     }
 
+    public function replyPost(){
```

Comment:
You should be modifying publishPost function rather than creating replyPost as it is the exact same other than `$parent_id`. Modify publishPost to use `$parent_id`.